crates/svm-runtime/src/vmcalls/storage.rs

```diff
@@ -5,11 +5,95 @@ use wasmer_runtime::Ctx as WasmerCtx;
 
 use svm_layout::VarId;
 
+macro_rules! store_n_impl {
+    ($nbytes:expr, $ctx:ident, $mem_idx:expr, $mem_ptr:expr, $var_id:expr) => {{
+        use crate::helpers;
+        use svm_layout::VarId;
+
+        let mem_ptr = $mem_ptr as usize;
+        let view = &$ctx.memory($mem_idx).view::<u8>()[mem_ptr..(mem_ptr + $nbytes)];
+
+        let bytes: Vec<u8> = view.iter().map(|cell| cell.get()).collect();
+
+        let storage = helpers::wasmer_data_app_storage($ctx.data);
+        storage.write_var(VarId($var_id), bytes);
+    }};
+}
+
+macro_rules! load_n_impl {
+    ($nbytes:expr, $ctx:ident, $var_id:expr, $mem_idx:expr, $mem_ptr:expr) => {{
+        use crate::helpers;
+        use svm_layout::VarId;
+
+        let storage = helpers::wasmer_data_app_storage($ctx.data);
+
+        let bytes = storage.read_var(VarId($var_id));
+        let nbytes = bytes.len();
+
+        assert_eq!(nbytes, $nbytes);
+
+        let mem_ptr = $mem_ptr as usize;
+        let view = &$ctx.memory($mem_idx).view::<u8>()[mem_ptr..(mem_ptr + $nbytes)];
+
+        for (cell, &byte) in view.iter().zip(bytes.iter()) {
+            cell.set(byte);
+        }
+    }};
+}
+
+/// Stores memory cells `[mem_ptr, mem_ptr + 1, ..., mem_ptr + 19]` into variable `var_id`.
+///
+/// # Panics
+///
+/// Panics if variable `var_id`'s length isn't 20 bytes.
+pub fn store160(ctx: &mut WasmerCtx, mem_idx: u32, mem_ptr: u32, var_id: u32) {
+    use_gas!("store160", ctx);
+
+    store_n_impl!(20, ctx, mem_idx, mem_ptr, var_id);
```

Comment:
The only difference between the `store160` and `store256` is for the `#nbytes`.

The other option here is to write a function and annotate it with `#[inline]`.
It will still be a bit less performant since than a macro since `nbytes` will be used as a local variable instead of the macro value injection (which occurs during compile-time).

Reply:
1. Why will `#[inline]` annotation be required?
1. So the the only purpose of using a macro here is to avoid the local variable(s) declaration/assignment?